dumoRecordsHandler/Interfaces/IContainer.h

```diff
@@ -0,0 +1,29 @@
+//
+// Created by emyrmak on 7/24/2019.
+//
+#pragma once
+
+#include "IRecord.h"
+#include "Student.h"
+#include <algorithm>
+#include <iostream>
+#include <fstream>
+#include <vector>
+
+
+class IContainer {
+public:
+    IContainer() = default;
+    ~IContainer() = default;
```

Comment:
And this is undefined behavior. If you use ANY virtual function in class, it's destructor has to be virtual.